src/components/ExchangePage/index.jsx

```diff
@@ -623,13 +624,18 @@ export default function ExchangePage({ initialCurrency, sending = false, params
       }
     }
 
-    let estimate, method, args, value
+    let estimate, method, args, value, ethTransactionSize
 
-    let inputEthPerToken = 1
-    if (inputCurrency !== 'ETH') {
-      inputEthPerToken = inputReserveToken && inputReserveETH ? inputReserveETH / inputReserveToken : null
+    if (inputCurrency === 'ETH') {
+      ethTransactionSize = inputValueFormatted
+    } else if (outputCurrency === 'ETH') {
+      ethTransactionSize = inputValueFormatted * amountFormatter(exchangeRate, 18, 6, false)
+    } else {
+      const tokenBalance = inputReserveToken && new BigNumber(inputReserveToken.toString())
+      const ethBalance = inputReserveETH && new BigNumber(inputReserveETH.toString())
+      let ethRate = ethBalance && tokenBalance && ethBalance.div(tokenBalance)
+      ethTransactionSize = inputValueFormatted * ethRate
```

Comment:
is this right?? feel like we should probably never be using `*`, and isn't `inputValueFormatted` a string?